h/search/index.py

```diff
@@ -26,7 +27,17 @@ class Window(namedtuple('Window', ['start', 'end'])):
     pass
 
 
-def index(es, annotation, request, target_index=None):
+def index(models_annotation, request):
+    search_annotation = persistence.Annotation.create(models_annotation)
+
+    request.registry.notify(AnnotationTransformEvent(
+        request, models_annotation, search_annotation.to_dict(),
+    ))
```

Comment:
`index_old()` passes a dict from `AnnotationSearchIndexPresenter` to `notify()`. I was kind of hoping to avoid `AnnotationSearchIndexPresenter` and be able to get rid of it, by using `elasticsearch_dsl`'s `to_dict()`. There's some custom stuff that we do in `AnnotationSearchIndexPresenter` and `DocumentSearchIndexPresenter` though (e.g. we extract the authority from the userid) so it won't be quite that simple in the end. I'm hoping we can add a custom `to_dict()` to `h.search.persistence.Annotation` to contain that logic though (and possibly add some properties to `h.models.Annotation` for it to use too) and still be able to get rid of `AnnotationSearchIndexPresenter` and `DocumentSearchIndexPresenter`.

(I don't like `AnnotationSearchIndexPresenter` and `DocumentSearchIndexPresenter` as components because all they do is return giant dicts, they can't be usefully tested in isolation because unit tests for those classes wouldn't test that those dicts are indexable into the Elasticsearch index with its mapping or that queries against the indexed documents work as expected. If annotations need to be turned into dicts for search indexing, that `to_dict()` code belongs in `h.search.persistence.Annotation`.)

Reply:
Also why is `index_old` passing a dict from `AnnotationSearchIndexPresenter` to `notify()`? `AnnotationSearchIndexPresenter` is supposed to be for presenting an annotation for search indexing, and not also for other purposes. What gets done with this dict that goes into `notify()`?

Reply:
I like the idea of adding properties to Annotation. Do you for-see cases where that still wouldn't enable us to get rid of the Presenter's?

Reply:
I'm thinking we won't need the presenters